File: backend/internal/models/order.go
package models

import "time"

type Order struct {
	Id             string    `json:"id" db:"id"`
	Count          int       `json:"count" db:"count"`
	OrderNumber    string    `json:"orderNumber" db:"order_number"`
	Notes          string    `json:"notes" db:"notes"`
	DateOfIssue    int       `json:"dateOfIssue" db:"date_of_issue"`
	DateOfDispatch int       `json:"dateOfDispatch" db:"date_of_dispatch"`
	DateOfAdoption int       `json:"dateOfAdoption" db:"date_of_adoption"`
	ClosingDate    int       `json:"closingDate" db:"closing_date"`
	Urgent         bool      `json:"urgent" db:"urgent"`
	Status         string    `json:"status" db:"status"`
	UpdatedAt      time.Time `json:"updatedAt" db:"updated_at"`
	CreatedAt      time.Time `json:"createdAt" db:"created_at"`
	Total          int       `json:"-" db:"total_count"`
}

type ImportantOrders struct {
	Urgent  int `json:"urgent" db:"urgent"`
	Overdue int `json:"overdue" db:"overdue"`
	Nearest int `json:"nearest" db:"nearest"`
}

type GetOrderDTO struct {
	Page    *Page
	Sort    []*Sort
	Filters []*Filter
}

type CreateOrderDTO struct {
	Id             string               `json:"id" db:"id"`
	OrderNumber    string               `json:"orderNumber" db:"order_number" binding:"required"`
	Notes          string               `json:"notes" db:"notes"`
	DateOfIssue    int                  `json:"dateOfIssue" db:"date_of_issue" binding:"required"`
	DateOfDispatch int                  `json:"dateOfDispatch" db:"date_of_dispatch"`
	Urgent         bool                 `json:"urgent" db:"urgent"`
	Status         string               `json:"status" db:"status"`
	Positions      []*CreatePositionDTO `json:"positions" binding:"required"`
}

type OrderDTO struct {
	Id             string    `json:"id" db:"id"`
	OrderNumber    string    `json:"orderNumber" db:"order_number" binding:"required"`
	Notes          string    `json:"notes" db:"notes"`
	DateOfIssue    int       `json:"dateOfIssue" db:"date_of_issue" binding:"required"`
	DateOfDispatch int       `json:"dateOfDispatch" db:"date_of_dispatch"`
	DateOfAdoption int       `json:"dateOfAdoption" db:"date_of_adoption"`
	ClosingDate    int       `json:"closingDate" db:"closing_date"`
	Urgent         bool      `json:"urgent" db:"urgent"`
	Status         string    `json:"status" db:"status"`
	UpdatedAt      time.Time `json:"updatedAt" db:"updated_at"`
}

type DeleteOrderDTO struct {
	Id string `json:"id" binding:"required"`
}

File: frontend/src/features/order/types/order.ts
import { IPosition, IPositionDTO } from '@/features/position/types/position'

export type Status = 'new' | 'processing' | 'closed'

export interface IOrder {
	id: string
	count: number
	orderNumber: string
	dateOfIssue: number
	dateOfDispatch: number
	dateOfAdoption: number
	closingDate: number
	urgent: boolean
	status: Status
	notes: string
	// positions: IPosition[]
}

export interface IOrderDTO {
	id?: string
	orderNumber: string
	dateOfIssue: number
	dateOfDispatch: number
	dateOfAdoption?: number
	closingDate?: number
	// dateOfExecution: number
	urgent: boolean
	notes: string
	status: Status
	positions?: IPositionDTO[]
}

export interface IEditOrderDTO extends IOrderDTO {
	positions?: IPosition[]
}

export interface IImportantOrders {
	urgent: number
	overdue: number
	nearest: number
}
